File: VB/T100846/Models/MyModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Namespace MVCxGridViewDataBinding.Models
	Public Class MyModel
		Public Sub New()
			Me.New(0)

		End Sub
		Public Sub New(ByVal index As Integer)
			ModelID = index
			ModelName = "Name" & index
			ModelState = index Mod 2 = 0
			ModelDate = DateTime.Now
			If index < 25 Then
				Quantity = index + 5 + index \2
			Else
				Quantity = index + 5 - index\2
			End If
		End Sub
		' Fields...
		Private _Quantity As Integer
		Private _ModelID As Integer
		Private _ModelName As String
		Private _ModelState? As Boolean
		Private _ModelDate As DateTime

		Public Property ModelID() As Integer
			Get
				Return _ModelID
			End Get
			Set(ByVal value As Integer)
				_ModelID = value
			End Set
		End Property
		Public Property ModelName() As String
			Get
				Return _ModelName
			End Get
			Set(ByVal value As String)
				_ModelName = value
			End Set
		End Property
		Public Property ModelState() As Boolean?
			Get
				Return _ModelState
			End Get
			Set(ByVal value? As Boolean)
				_ModelState = value
			End Set
		End Property
		Public Property ModelDate() As DateTime
			Get
				Return _ModelDate
			End Get
			Set(ByVal value As DateTime)
				_ModelDate = value
			End Set
		End Property

		Public Property Quantity() As Integer
			Get
				Return _Quantity
			End Get
			Set(ByVal value As Integer)
				_Quantity = value
			End Set
		End Property
		Public Shared Function GetModelList() As List(Of MyModel)
			Dim l As New List(Of MyModel)()
			For i As Integer = 0 To 49
				l.Add(New MyModel(i))
			Next i
			Return l
		End Function
		Public Shared Sub UpdateModel(ByVal l As List(Of MyModel), ByVal m As MyModel)

			Dim editedModel As MyModel = l.Where(Function(x) x.ModelID = m.ModelID).First()
			editedModel.ModelName = m.ModelName
			editedModel.ModelDate = m.ModelDate
			editedModel.ModelState = m.ModelState
		End Sub
		Public Shared Sub DeleteModel(ByVal l As List(Of MyModel), ByVal mID As Integer)
			l.Remove(l.Where(Function(x) x.ModelID = mID).First())
		End Sub
		Public Shared Sub InsertModel(ByVal l As List(Of MyModel), ByVal m As MyModel)
			m.ModelID = l.Count + 1
			l.Add(m)
		End Sub
	End Class
End Namespace

File: CS/CustomSummary/Models/MyModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace MVCxGridViewDataBinding.Models
{
    public class MyModel
    {
        public MyModel()
            : this(0)
        {

        }
        public MyModel(int index)
        {
            ModelID = index;
            ModelName = "Name" + index;
            ModelState = index % 2 == 0;
            ModelDate = DateTime.Now;
            if(index < 25)
                Quantity = index + 5  +  index /2;
            else
                Quantity = index + 5 - index/2;
        }
        // Fields...
        private int _Quantity;
        private int _ModelID;
        private string _ModelName;
        private Boolean? _ModelState;
        private DateTime _ModelDate;

        public int ModelID
        {
            get { return _ModelID; }
            set { _ModelID = value; }
        }
        public string ModelName
        {
            get { return _ModelName; }
            set { _ModelName = value; }
        }
        public Boolean? ModelState
        {
            get { return _ModelState; }
            set { _ModelState = value; }
        }
        public DateTime ModelDate
        {
            get { return _ModelDate; }
            set { _ModelDate = value; }
        }

        public int Quantity
        {
            get { return _Quantity; }
            set { _Quantity = value; }
        }
        public static List<MyModel> GetModelList()
        {
            List<MyModel> l = new List<MyModel>();
            for (int i = 0; i < 50; i++)
            {
                l.Add(new MyModel(i));
            }
            return l;
        }
        public static void UpdateModel(List<MyModel> l, MyModel m)
        {

            MyModel editedModel = l.Where(x => x.ModelID == m.ModelID).First();
            editedModel.ModelName = m.ModelName;
            editedModel.ModelDate = m.ModelDate;
            editedModel.ModelState = m.ModelState;
        }
        public static void DeleteModel(List<MyModel> l, int mID)
        {
            l.Remove(l.Where(x => x.ModelID == mID).First());
        }
        public static void InsertModel(List<MyModel> l, MyModel m)
        {
            m.ModelID = l.Count + 1;
            l.Add(m);
        }
    }
}